File: Codeforces Round #886 (Div. 4)/F.cpp
/* Code By [ junah ]
GitHub : [ junah201 ] */
#include <bits/stdc++.h>
using namespace std;
#define all(v) v.begin(), v.end()
#define int long long int
#define INF LONG_LONG_MAX

int T;

int solve()
{
    int N;
    scanf("%lld", &N);

    vector<int> A(N);
    vector<int> ans(N + 1, 0);
    for (int i = 0; i < N; i++)
    {
        scanf("%lld", &A[i]);
    }

    for (auto i : A)
    {
        for (int j = i; j <= N; j += i)
        {
            ans[j] += 1;
        }
    }

    int result = 0;
    for (auto i : ans)
    {
        result = max(result, i);
    }
    return result;
}

signed main()
{
    scanf("%lld", &T);
    while (T--)
    {
        printf("%lld\n", solve());
    }

    return 0;
}

File: Codeforces Round #886 (Div. 4)/F.py
import sys

input = sys.stdin.readline

T = int(input())


def solve():
    N = int(input())
    A = [i for i in list(map(int, input().split())) if i <= N]

    ans = [
        0 for _ in range(N+1)
    ]

    for i in A:
        init = i
        while i <= N:
            ans[i] += 1
            i += init

    print(max(ans))


for _ in range(T):
    solve()
